### ISSUE TITLE
[BUG]: llama2 hybrid_parallel or 3d giving None loss when using pp_size > 1

### ISSUE THREAD

**Issue by PurvangL** [hpcaitech/ColossalAI#5683]
### 🐛 Describe the bug

Hi, I am trying to run llama2 7B model on [yizhongw/self_instruct](https://huggingface.co/datasets/yizhongw/self_instruct) dataset. As title suggest, training with hybrid_parallel or 3d plugin giving None loss, but other plugin works as expected without any issue.

tp_size=4
pp_size=2

```
torchrun --nproc_per_node=8 --nnodes=1 --node_rank=0 --rdzv_id=456 --rdzv_backend=c10d --rdzv_endpoint=x.x.x.x:x llama2/finetune.py --plugin "hybrid_parallel" --dataset "yizhongw/self_instruct" --model_path /workspace/ColossalAI/llama2-7b-hf/ --task_name "super_natural_instructions" --max_length 512 -e 10 -b 64 --lr  0.00002 --grad_checkpoint --mixed_precision bf16
```

### Environment

Docker image: [nvcr.io/nvidia/pytorch:23.12-py3](http://nvcr.io/nvidia/pytorch:23.12-py3)
colossalai==0.3.6

**Comment by Edenzzzz** [comment]
Hi,
Could you try pulling the latest main branch? I don't have trouble running pp_size = 2.

**Comment by PurvangL** [comment]
Hi, 
Unfortunately latest main branch doesn't have finetune.py file any more. Also, benchmark.py doesn't support huggingface dataset and it runs with random dataset. 
I want to run it on mentioned dataset with llama2 model.

**Comment by Edenzzzz** [comment]
I think it should support any dataset. Have you tried replacing the random dataset with this one-liner from huggingface? https://github.com/hpcaitech/ColossalAI/blob/8020f4263095373e4c7ad1b15e54b966a8ccb683/examples/language/llama2/finetune.py#L209

**Comment by PurvangL** [comment]
I also did try replacing and modifying code. I still got None loss. 
Did you get valid loss value?

**Comment by Edenzzzz** [comment]
Actually, in pipeline parallel only the last stage outputs loss, so this is not a bug. You'll need to do this to see the actual loss.
<img width="996" alt="image" src="https://github.com/hpcaitech/ColossalAI/assets/87317405/85fe6374-9ad7-4309-ae0f-5376598f2b35">

Also, there's a llama fine-tuning example in applications/Colossal-LLaMA.

**Comment by PurvangL** [comment]
Thanks @Edenzzzz . That's solves the issue I had.
